.github/workflows/build_ncs.yml

```diff
@@ -0,0 +1,64 @@
+name: Build binary for Nordic NCS
+
+# Runs when pushed to main, semver tagged release or any PR
+on:
+  push:
+    branches: [ main ]
+    tags: [ 'v*.*.*' ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    # Uses matrix to test combinations. Update when SDK changes or to add new boards.
+    strategy:
+      matrix:
+        ZEPHYR_SDK_VERSION: [0.16.1]
+        BOARD: ["nrf9160dk_nrf9160_ns","aludel_mini_v1_sparkfun9160_ns"]
+
+    steps:
+      - name: Install OS dependencies
+        run: >
+          sudo apt-get install -y
+          device-tree-compiler
+          git
+          ninja-build
+          python3
+          python3-pip
+          python3-wheel
+          wget
+          xz-utils
+
+      - name: Install build tools
+        run: |
+          pip3 install wheel
+          pip3 install cmake west
+
+      - name: Install Zephyr SDK
+        run: |
+          wget -q "https://github.com/zephyrproject-rtos/sdk-ng/releases/download/v${{ matrix.ZEPHYR_SDK_VERSION }}/zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz"
+          mkdir -p /opt/zephyr-sdk
+          tar -xvf zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz -C /opt/zephyr-sdk --strip-components=1
+          /opt/zephyr-sdk/setup.sh -t arm-zephyr-eabi
+          rm zephyr-sdk-${{ matrix.ZEPHYR_SDK_VERSION }}_linux-x86_64_minimal.tar.xz
+
+      - name: Build with West
+        run: |
+          west init -m https://github.com/$GITHUB_REPOSITORY .
+          west update
+          west zephyr-export
+          pip3 install -r deps/zephyr/scripts/requirements.txt
+          west build -p -b ${{ matrix.BOARD }} app -- -DCONFIG_MCUBOOT_IMAGE_VERSION=\"0.0.0\"
+          mv build/zephyr/zephyr.bin build/zephyr/${{ matrix.BOARD }}.bin
+          mv build/zephyr/zephyr.elf build/zephyr/${{ matrix.BOARD }}.elf
+
+      # Run IDs are unique per repo but are reused on re-runs
+      - name: Save artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: build_artifacts_${{ github.run_id }}
+          path: |
+            build/zephyr/${{ matrix.BOARD }}.bin
+            build/zephyr/${{ matrix.BOARD }}.elf
```

Comment:
As mentioned above... I would love a future improvement that triggers when a release is created and uploads the generated binaries automatically. That would be :fire: and a huge time saver.

Reply:
I've seen other projects treat `releases` as a separate process. I think Zephyr creates the release and CI later uploads artifacts once they're built.

Reply:
Ah, I always wondered how that worked. Still, whatever we can do to automate this would be useful. I get nervous making sure I'm renaming the correct files to the correct names and uploading. There's lots of room there for error.

Reply:
Added manual trigger. We should discuss release process and automation as a setup PR.